waspc/src/Wasp/Generator/Valid/PackageJson.hs

```diff
@@ -3,172 +3,183 @@ module Wasp.Generator.Valid.PackageJson
-
-validateDevelopmentDependencies :: P.PackageJson -> [GeneratorError]
-validateDevelopmentDependencies packageJson =
-  concat
-    [ validateDevelopment ("vite", show viteVersion),
-      validateDevelopment ("prisma", show prismaVersion)
-    ]
+    validateFile =
+      withFileName "package.json" $
+        V.validateAll $
+          [validateWorkspaces]
+            <> validateRuntimeDeps
+            <> validateDevDeps
+            <> validateOptionalDeps
+
+    validateRuntimeDeps =
+      validateRequiredDependency Runtime
+        <$> [ ("wasp", "file:.wasp/out/sdk/wasp"),
+              ("react-router-dom", show D.reactRouterVersion),
+              -- Installing the wrong version of "react-router-dom" can make users believe that they
+              -- can use features that are not available in the version that Wasp supports.
+              ("react", show D.reactVersion),
+              ("react-dom", show D.reactVersion)
+            ]
+
+    validateDevDeps =
+      validateRequiredDependency Development
+        <$> [ ("vite", show D.viteVersion),
+              ("prisma", show prismaVersion)
+            ]
+
+    validateOptionalDeps =
+      validateOptionalDependency
+        <$> [ ("typescript", show typescriptVersion),
+              ("@types/react", show D.reactTypesVersion),
+              ("@types/express", show expressTypesVersion)
+            ]
+
+-- | Validates that the 'workspaces' property in package.json includes all
+-- expected workspaces used by Wasp.
+validateWorkspaces :: P.PackageJson -> Validation ()
+validateWorkspaces =
+  valueOfField ("workspaces", P.workspaces) $ \case
+    Just workspaces -> allWorkspacesIncluded workspaces
+    Nothing -> noWorkspacesPropertyError
   where
-    validateDevelopment packageSpec = validatePackageJsonDependency packageJson packageSpec RequiredDevelopment
-
-validateOptionalDependencies :: P.PackageJson -> [GeneratorError]
-validateOptionalDependencies packageJson =
-  concat
-    [ validateOptional ("typescript", show typescriptVersion),
-      validateOptional ("@types/react", show reactTypesVersion),
-      validateOptional ("@types/express", show expressTypesVersion)
-    ]
+    allWorkspacesIncluded :: [String] -> Validation ()
+    allWorkspacesIncluded =
+      validateAll_ $ workspaceIncluded <$> expectedWorkspaces
+
+    workspaceIncluded :: String -> [String] -> Validation ()
+    workspaceIncluded expectedWorkspace actualWorkspaces
+      | expectedWorkspace `elem` actualWorkspaces = pure ()
+      | otherwise = makeMissingWorkspaceError expectedWorkspace
+
+    makeMissingWorkspaceError expectedWorkspace =
+      failure $
+        "Wasp requires "
+          ++ show expectedWorkspace
+          ++ " to be included."
+
+    noWorkspacesPropertyError =
+      failure $
+        "Wasp requires the value "
+          ++ show expectedWorkspaces
+          ++ "."
+
+    expectedWorkspaces = S.toList NW.workspaceGlobs
```

Comment:
I know this was not done by you in this PR, but `workspaceGlobs` name here is lacking -> I looked up its definition, and if I was reviewing that code, I would ask it be named `requiredWorkspaceGlobs`. Maybe this is opportunity to rectify this. Because I was reading this line and I was thinking, hm ok, why are expectedWorkspaces matching some workspaceGlobs? How are those connected, why are those same as expectedWorkspaces? Is there some kind of assumption being made?

But then I went to definition of workspaceGlobs and I see in the comment that it says they are required. Why is that only in the comment and not in the name, I don't know, but I think we should improve that.